src/services/pathfinding.service.ts

```diff
@@ -35,24 +36,46 @@ class PathFindingService {
    *
    * @returns The list of paths to travese between the start and end POI's
    */
-  public findPathBetweenPOIs = (start: POI, end: POI): FloorPath[] | null => {
+  public findPathBetweenPOIs = (
+    start: POI,
+    end: POI,
+    accessibilityMode: boolean
+  ): FloorPath[] | null => {
     if (start.buildingId === end.buildingId) {
-      return this.findBuildingPath(start, end);
+      return this.findBuildingPath(start, end, accessibilityMode);
     }
-
-    const startBuildingExit = POIInfo.find(
-      ({ category, buildingId }) =>
-        category === POICategory.Exit && buildingId === start.buildingId
-    ) as ConnectorPOI;
-    const endBuildingExit = POIInfo.find(
-      ({ category, buildingId }) =>
-        category === POICategory.Exit && buildingId === end.buildingId
-    ) as ConnectorPOI;
+    const startBuildingExit = this.findPOI(POICategory.Exit, start.buildingId);
+    const endBuildingExit = this.findPOI(POICategory.Exit, end.buildingId);
 
     // TODO: connect these paths to the outdoor path.
-    return this.findBuildingPath(start, startBuildingExit).concat(
-      this.findBuildingPath(end, endBuildingExit)
+    const startBuildingPath = this.findBuildingPath(
+      start,
+      startBuildingExit,
+      accessibilityMode
     );
+    const endBuildingPath = this.findBuildingPath(
+      end,
+      endBuildingExit,
+      accessibilityMode
+    );
```

Comment:
Could you change this to:
```
const endBuildingPath = this.findBuildingPath(
      endBuildingExit,
      end,
      accessibilityMode
    );
```
We found out that this would give a more accurate route in terms of ordering